tools/src/main/scala/scala/scalanative/tools/Config.scala

```diff
@@ -1,80 +1,197 @@
 package scala.scalanative
 package tools
 
-import java.io.File
+import java.nio.file.{Path, Paths}
+
 import nir.Global
 
 sealed trait Config {
 
+  /** Path to the nativelib jar. */
+  def nativeLib: Path
+
+  /** The driver to use for the optimizer. */
+  def driver: OptimizerDriver
```

Comment:
Having customizable optimizer driver implies leaking NIR into the API. If it's not customizable it might as well be removed from the config object (and materialized on demand in the build api implementation).

Reply:
With `passes` field commented out, it's not leaking NIR any more, but referencing to optimizer-internal class in a public API is still suboptimal.

Reply:
Is it okay as it is now?